roles/keystone-setup/tasks/keystone-idp.yml

```diff
@@ -0,0 +1,10 @@
+- name: Register keystone sp info on keystone idp
+  k2k_identity_provider: auth_url="http://127.0.0.1:{{ endpoints.keystone_admin.port.backend_api }}/v3"
+                         username="admin"
+                         password="{{ secrets.admin_password }}"
+                         project_name="admin"
+                         domain_name="Default"
+                         service_provider_id="{{ item.id }}"
+                         service_provider_url="{{ item.service_provider_url }}"
+                         service_provider_auth_url="{{ item.service_provider_auth_url }}"
+  with_items: "{{ keystone.keystone_identity_provider.service_providers }}"
```

Comment:
Particularly with a lot of arguments like this, I'd prefer the task use the syntax like:
```
- name: Register keystone sp info on keystone idk
  k2k_identity_provider:
    auth_url: "http://127.0.0.1:{{ endpoints.keystone_admin.port.backend_api }}/v3"
    password: "{{ secrets.admin_password }}"
    project_name: "admin"
```